server/models/user.js

```diff
@@ -37,22 +37,37 @@ UserSchema.plugin(uniqueValidator, {
 UserSchema.pre('save', function(next) {
   const password = this.password;
   if (password) {
-    bcrypt.hash(password, 10)
-      .then((hash) => {
+    bcrypt
+      .hash(password, 10)
+      .then(hash => {
         this.password = hash;
         next();
       })
-      .catch((err) => next(err))
+      .catch(err => next(err));
   } else {
     next();
   }
 });
 
+UserSchema.statics.authenticate = function(email, password, callback) {
+  User.findOne({ email }, function(error, user) {
+    if (error) {
+      return callback(error);
```

Comment:
This worries me that we'll be returning a more sensitive error. I'm curious what kind of error would result from this. I wonder if we should either test it and find out, or let this produce a 500 error (don't pass it through the callback until we test it).

Reply:
Removed.